File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float change;
    do
    {
        change = get_float("Change owed: ");
    } while (change <= 0);

    int cents = round(change * 100);
    int monedasPosibles[] = {25, 10, 5, 1};
    int contador = 0;
    for (int i = 0; i < 4; i++)
    {
        while (cents - monedasPosibles[i] >= 0)
        {
            cents -= monedasPosibles[i];
            contador++;
        }
        if (cents == 0)
        {
            break;
        }
    }
    printf("%i\n", contador);
}

File: pset6/cashpy/cash.py
 
from cs50 import get_float
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

cents = round(change * 100)
monedasPosibles = [25, 10, 5, 1]
counter = 0
for i in range(4):
    while cents - monedasPosibles[i] >= 0:
        cents -= monedasPosibles[i]
        counter += 1
    if cents == 0:
         break
print(counter)
